server/src/main/java/com/google/fhir/proxy/BearerAuthorizationInterceptor.java

```diff
@@ -268,21 +268,21 @@ public boolean authorizeRequest(RequestDetails requestDetails) {
     logger.debug("Authorized request path " + requestPath);
     try {
       HttpResponse response = fhirClient.handleRequest(servletDetails);
-      // TODO pass along the response to the client in case of errors (b/211233113).
-      HttpUtil.validateResponseEntityOrFail(response, requestPath);
       // TODO communicate post-processing failures to the client (b/211243404).
       String content = null;
-      try {
-        // For post-processing rationale/example see b/207589782#comment3.
-        content = outcome.postProcess(response);
-      } catch (Exception e) {
-        // Note this is after a successful fetch/update of the FHIR store. That success must be
-        // passed to the client even if the access related post-processing fails.
-        logger.error(
-            "Exception in access related post-processing for {} {}",
-            requestDetails.getRequestType(),
-            requestDetails.getRequestPath(),
-            e);
+      if (HttpUtil.isResponseEntityValid(response)) {
+        try {
+          // For post-processing rationale/example see b/207589782#comment3.
+          content = outcome.postProcess(response);
+        } catch (Exception e) {
+          // Note this is after a successful fetch/update of the FHIR store. That success must be
```

Comment:
nit: please update this comment as this is not necessarily "successful" anymore.

Reply:
This will only be if the response from the FHIR server is a successful response as specified by the if condition at line 273.

Reply:
You are right, I misunderstood the logic here. But now I think there is another issue: With this change it is possible to get a `null` `entity`  on line 288 below. So it is probably better to account for that. IOW, we still want to throw our exception if `response.getEntity()` is `null` but let a 400 response to go through. Makes sense?

Reply:
Yes! Thank you pointing out, made changes to check for the existence of the response entity